encoding/protobq/schema.go

```diff
@@ -24,6 +24,10 @@ type SchemaOptions struct {
 	// UseOneofFields adds an extra STRING field for oneof fields with the name of the oneof,
 	// containing the name of the field that is set.
 	UseOneofFields bool
+	// UseOptionalFields causes any field not marked "optional" in the proto to be a REQUIRED field.
+	// Lists and Maps are always NULLABLE as proto3 does not seem to support optional for those types.
```

Comment:
Aren't messages also always nullable?

Since proto3 optional is only meant to apply to primitives, I think we're misusing it here by interpreting `not optional` to mean `required`.

From the release notes when `optional` was released:

> [experimental] Singular, non-message typed fields in proto3 now support
> presence tracking. This is enabled by adding the "optional" field label

I.e. I don't think the absence of optional is meant to imply `required`. Just that it's not important to tell the difference between the zero value and an absent value.

Reply:
We're talking about translating protobuf to bigquery, we can technnically define what optional means and doesn't mean when the plugin is translating to a BigQuery schema so the proto definition doesn't necessarily apply.

While it is true that technically (at least not in Go) there is no difference between an optional message and a non-optional since it all becomes a pointer, protoc does parse it. We did start using it recently to signal "This is optional" but maybe it was never intended to be used like that. The documentation lacks a bit, this says that optional is compatible with `repeated` but protoc doesn't like it

https://developers.google.com/protocol-buffers/docs/proto3#updating

While I think that, for those wanting to make use of BQ Required, this PR makes sense, I think our discussions outside of this PR make sense to avoid BQ Required for the same reasons proto3 avoids that so I am half tempted to close this PR and call it done. Thoughts?